lsp-mode.el

```diff
@@ -1059,14 +1059,61 @@ every element of it is of type string, else nil."
    (vectorp candidate)
    (seq-every-p #'stringp candidate)))
 
+(make-obsolete 'lsp--string-vector-p nil "7.0.2")
+
+(defun lsp-editable-vector-match (widget value)
+  "Function for `lsp-editable-vector' :match."
+  ;; Value must be a list or a vector and all the members must match the type.
+  (and (or (listp value) (vectorp value))
+       (length (cdr (lsp-editable-vector-match-inline widget value)))))
+
+(defun lsp-editable-vector-match-inline (widget value)
+  "Value for `lsp-editable-vector' :match-inline."
+  (let ((type (nth 0 (widget-get widget :args)))
+        (ok t)
+        found)
+    (while (and value ok)
+      (let ((answer (widget-match-inline type value)))
+        (if answer
+            (let ((head (if (vectorp answer) (aref answer 0) (car answer)))
+                  (tail (if (vectorp answer) (seq-drop 1 answer) (cdr answer))))
+              (setq found (append found head)
+                    value tail))
+          (setq ok nil))))
+    (cons found value)))
+
+(define-widget 'lsp-editable-vector 'editable-list
+  "A subclass of `editable-list' that accepts and returns a
+vector instead of a list."
+  :value-to-external 'lsp-editable-vector-value-to-external
+  :value-to-internal 'lsp-editable-vector-value-to-internal
+  :match 'lsp-editable-vector-match
+  :match-inline 'lsp-editable-vector-match-inline)
+
+(defun lsp-editable-vector-value-to-external (_widget internal-value)
+  "Convert the internal list value to a vector."
+  (if (listp internal-value)
+      (apply 'vector internal-value)
+    internal-value))
+
+(defun lsp-editable-vector-value-to-internal (_widget external-value)
+  "Convert the external vector value to a list."
+  (if (vectorp external-value)
+      (append external-value nil)
+    external-value))
+
+(define-widget 'lsp-repeatable-vector 'lsp-editable-vector
```

Comment:
Except `lsp-repeatable-vector`, I think other related functions, widget can be private, WDYT? Can you add double dash to their name?

Reply:
No, this type is very much meant for consumption by the clients. Conceivably there will be configurations that look like `[1, 3, 5]`, if that's the case client authors can just use `(lsp-repeatable-vector integer)` as the defcustom type instead of asking us to make a `lsp-integer-vector` or some weird thing. `lsp-string-vector` is only preserved for backward compatibility. I actually think we should deprecate that.

Reply:
That why I said except `lsp-repeatable-vector`. Now everything you added has public scope naming.